src/main/java/tutu/user/Parser.java

```diff
@@ -0,0 +1,32 @@
+package tutu.user;
+
+import tutu.exception.InvalidInputException;
+
+public class Parser {
+
+    public enum Function {
+        LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, FIND
+    }
+
+    public static Function checkFunction(String cmd) throws InvalidInputException {
+        if (cmd.equals("list")) {
+            return Function.LIST;
+        } else if (cmd.length() > 3 && cmd.substring(0, 4).equals("mark")) {
+            return Function.MARK;
+        } else if (cmd.length() > 5 && cmd.substring(0, 6).equals("unmark")) {
+            return Function.UNMARK;
+        } else if (cmd.length() > 3 && cmd.substring(0, 4).equals("todo")) {
+            return Function.TODO;
+        } else if (cmd.length() > 7 && cmd.substring(0, 8).equals("deadline")) {
+            return Function.DEADLINE;
+        } else if (cmd.length() > 4 && cmd.substring(0, 5).equals("event")) {
+            return Function.EVENT;
+        } else if (cmd.length() > 5 && cmd.substring(0, 6).equals("delete")) {
+            return Function.DELETE;
+        } else if (cmd.length() > 4 && cmd.substring(0, 4).equals("find")) {
+            return Function.FIND;
+        } else {
+            throw new InvalidInputException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
```

Comment:
Perhaps a switch statement would be neater. You could also consider splitting instead of substring. E.g.
```java
String[] splitSpace = input.split(" ");
String cmd = splitSpace[0];

switch (cmd) {
case "mark":
...
```